calm_adapter/calm_api_client/src/test/scala/uk/ac/wellcome/platform/calm_api_client/CalmApiClientTest.scala

```diff
@@ -0,0 +1,155 @@
+package uk.ac.wellcome.platform.calm_api_client
+
+import java.time.LocalDate
+
+import akka.http.scaladsl.model.HttpResponse
+import akka.http.scaladsl.model.headers.{
+  Authorization,
+  BasicHttpCredentials,
+  Cookie,
+  RawHeader
+}
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.funspec.AnyFunSpec
+import org.scalatest.matchers.should.Matchers
+import uk.ac.wellcome.akka.fixtures.Akka
+import uk.ac.wellcome.fixtures.TestWith
+import uk.ac.wellcome.platform.calm_api_client.fixtures.{
+  CalmHttpTestClient,
+  CalmResponseGenerators
+}
+
+import scala.concurrent.ExecutionContext.Implicits.global
+
+class CalmApiClientTest
+    extends AnyFunSpec
+    with Matchers
+    with ScalaFutures
+    with CalmResponseGenerators
+    with Akka {
+
+  val url = "calm.api"
+  val username = "calm-user"
+  val password = "calm-password"
+  val query = CalmQuery.ModifiedDate(LocalDate.of(2000, 1, 1))
+  val suppressedField = "Secret" -> "Shhhh"
+
+  it("performs search requests") {
+    val nResults = 10
+    val responses = List(searchResponse(nResults))
+    withApiClient(responses) {
+      case (apiClient, _) =>
+        whenReady(apiClient.search(query)) { response =>
+          response shouldBe CalmSession(nResults, Cookie(cookie))
+        }
+    }
+  }
+
+  it("performs summary requests") {
+    val responses = List(summaryResponse(List("RecordID" -> "1")))
+    withApiClient(responses) {
+      case (apiClient, _) =>
+        whenReady(apiClient.summary(1)) { response =>
+          response shouldBe CalmRecord(
+            "1",
+            Map("RecordID" -> List("1")),
+            retrievedAt)
+        }
+    }
+  }
+
+  it("uses basic auth credentials for requests") {
+    val responses =
+      List(searchResponse(1), summaryResponse(List("RecordID" -> "1")))
```

Comment:
I'd prefer to see keyword arguments throughout here, so it's clear what this number means – it's not obvious that it's number of results rather than, say, the ID.